docs/spec/sources.md

```diff
@@ -362,5 +376,37 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
+
+- `name` is the [Source Custom Object](#source-custom-objects) name
+- `plural` is the [Source CRD](#source-crds)
+  [plural](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `group` is the [Source CRD](#source-crds)
+  [API group](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+
+Examples:
+
+- `hello.pingsources.sources.knative.dev`
+- `rabbitmq-source.rabbitmqsources.sources.knative.dev`
```

Comment:
That example sounds confusing to me because `name` repeats parts of both the `plural` and the `group`. IMO examples in a spec should be _extra careful_ about not confusing the reader in any manner.

```suggestion
- `myqueue.rabbitmqsources.sources.knative.dev`
```

Reply:
That's real: https://github.com/knative-sandbox/eventing-rabbitmq/blob/master/samples/source/event-source.yaml :-)

Changed.